azure-iot-device/samples/advanced-hub-scenarios/register_symmetric_key_group.py

```diff
@@ -0,0 +1,53 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import os
+import asyncio
+import base64
+import hmac
+import hashlib
+from azure.iot.device.aio import ProvisioningDeviceClient
+
+provisioning_host = os.getenv("PROVISIONING_HOST")
+id_scope = os.getenv("PROVISIONING_IDSCOPE")
+# This is the enrollment group name
+registration_id = os.getenv("PROVISIONING_GROUP_REGISTRATION_ID")
+# This is the group symmetric key
+master_symmetric_key = os.getenv("PROVISIONING_MASTER_SYMMETRIC_KEY")
+
+
+def derive_device_key():
```

Comment:
Can you put a comment at above this line that explains how you derive the key.  You may be able to take it straight from the dps documentation page.

Reply:
Just read the docs, and it's not that great for comments.  If you just spelled out what you're doing here it may be helpful.

---
In reply to: [337213080](https://github.com/Azure/azure-iot-sdk-python/pull/360#discussion_r337213080) [](ancestors = 337213080)